src/Core/src/Platform/iOS/WebViewExtensions.cs

```diff
@@ -55,7 +55,32 @@ public static void UpdateGoForward(this WKWebView platformWebView, IWebView webV
 
 		public static void UpdateReload(this WKWebView platformWebView, IWebView webView)
 		{
-			platformWebView?.Reload();
+			if (platformWebView == null)
+			{
+				return;
+			}
+
+			// Check if we have a source and platformWebView implements IWebViewDelegate
+			if (webView.Source != null && platformWebView is IWebViewDelegate webViewDelegate)
+			{
+				// If the source is an HTML source, we need to reload from the source
+				// since WKWebView.Reload() doesn't work properly with LoadHtmlString
+				var sourceType = webView.Source.GetType().Name;
+				if (string.Equals(sourceType, "HtmlWebViewSource", StringComparison.Ordinal))
```

Comment:
Could you eliminate string allocation and comparison by using direct type checking?

`if (webView.Source is HtmlWebViewSource)`

Reply:
@jsuarezruiz, We can't check the type directly because HtmlWebViewSource belongs to the Controls project, whereas the Extensions are defined in the Core project. So accessed the type using GetType().Name.